src/plugins/discover/public/embeddable/search_embeddable_component.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { I18nProvider } from '@osd/i18n/react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { SearchProps } from './search_embeddable';
+import {
+  DataGridTable,
+  DataGridTableProps,
+} from '../application/components/data_grid/data_grid_table';
+import { VisualizationNoResults } from '../../../visualizations/public';
+
+interface SearchEmbeddableProps {
+  searchProps: SearchProps;
+}
+export interface DiscoverEmbeddableProps extends DataGridTableProps {
+  totalHitCount: number;
+}
+
+export const DataGridTableMemoized = React.memo((props: DataGridTableProps) => (
+  <DataGridTable {...props} />
+));
+
+export function SearchEmbeddableComponent({ searchProps }: SearchEmbeddableProps) {
+  const discoverEmbeddableProps = {
+    columns: searchProps.columns,
+    indexPattern: searchProps.indexPattern,
+    onAddColumn: searchProps.onAddColumn,
+    onFilter: searchProps.onFilter,
+    onRemoveColumn: searchProps.onRemoveColumn,
+    onSort: searchProps.onSort,
+    rows: searchProps.rows,
+    onSetColumns: searchProps.onSetColumns,
+    sort: searchProps.sort,
+    displayTimeColumn: searchProps.displayTimeColumn,
+    services: searchProps.services,
+    totalHitCount: searchProps.totalHitCount,
+  } as DiscoverEmbeddableProps & { totalHitCount: number };
```

Comment:
```suggestion
  const discoverEmbeddableProps: DiscoverEmbeddableProps = {
    columns: searchProps.columns,
    indexPattern: searchProps.indexPattern,
    onAddColumn: searchProps.onAddColumn,
    onFilter: searchProps.onFilter,
    onRemoveColumn: searchProps.onRemoveColumn,
    onSort: searchProps.onSort,
    rows: searchProps.rows,
    onSetColumns: searchProps.onSetColumns,
    sort: searchProps.sort,
    displayTimeColumn: searchProps.displayTimeColumn,
    services: searchProps.services,
    totalHitCount: searchProps.totalHitCount,
  };
```
The difference is that this way, typescript ensures that `discoverEmbeddableProps` conforms to `DiscoverEmbeddableProps` and does not forceable change the type of your object to `DiscoverEmbeddableProps`.

Also `DiscoverEmbeddableProps` already has `totalHitCount` as defined on line 20